File: ATLASDEV/AtlasModel/Enterprise/TTAttributeDef.cs
using AtlasModel.Enterprise.Enums;
using System;

namespace AtlasModel.Enterprise
{
    public partial class TTAttributeDef
    {
        public string AttributeDefId { get; set; }
        public Guid? InterfaceDefId { get; set; }
        public string Body { get; set; }
        public string CheckBody { get; set; }
        public AttributeTargetEnum Target { get; set; }
        public AttributeEventTypeEnum EventType { get; set; }
        public AttributeWhenEnum When { get; set; }
    }
}

File: AtlasWeb/Core/wwwroot/app/NebulaClient/StorageManager/DefinitionManagement/TTAttributeDef.ts
import { Guid } from "../../Mscorlib/Guid";
import { TTObjectSubTypeEnum } from "../../Utils/Enums/TTObjectSubTypeEnum";
import { TTInterfaceDef } from "./TTInterfaceDef";
import { AttributeTargetEnum } from "../../Utils/Enums/AttributeTargetEnum";
import { AttributeEventTypeEnum } from "../../Utils/Enums/AttributeEventTypeEnum";
import { AttributeWhenEnum } from "../../Utils/Enums/AttributeWhenEnum";
import { TTDefCollection } from "./TTDefCollection";
import { TTAttributeParameterDef } from "./TTAttributeParameterDef";

export class TTAttributeDef {
    public SubType: TTObjectSubTypeEnum;
    public InterfaceDef: TTInterfaceDef;
    public InterfaceDefID: Guid;
    public Body: string;
    public CheckBody: string;
    public Target: AttributeTargetEnum;
    public EventType: AttributeEventTypeEnum;
    public When: AttributeWhenEnum;
    public AllParameterDefs: TTDefCollection<TTAttributeParameterDef>;
    public CanBeRun(target: AttributeTargetEnum, eventType: AttributeEventTypeEnum, when: AttributeWhenEnum): boolean {
        return null;
    }
}
